src/storage/storage_structure/lists/lists_update_store.cpp

```diff
@@ -171,54 +161,150 @@ void ListsUpdateStore::readValues(
+                            .at(nodeOffset);
+    ftOfInsertedRels->lookup(vectorsToRead, columnsToRead, listUpdates->insertedRelsTupleIdxInFT,
         listHandle.getStartElemOffset(), numTuplesToRead);
     valueVector->state->originalSize = numTuplesToRead;
 }
 
-bool ListsUpdateStore::hasAnyDeletedRelsInPersistentStore(
+bool ListsUpdatesStore::hasAnyDeletedRelsInPersistentStore(
     ListFileID& listFileID, node_offset_t nodeOffset) const {
-    auto relNodeTableAndDir = getRelNodeTableAndDirFromListFileID(listFileID);
-    auto& listUpdatesPerChunk = listUpdatesPerTablePerDirection[relNodeTableAndDir.dir].at(
-        relNodeTableAndDir.srcNodeTableID);
-    auto chunkIdx = StorageUtils::getListChunkIdx(nodeOffset);
-    if (!listUpdatesPerChunk.contains(chunkIdx) ||
-        !listUpdatesPerChunk.at(chunkIdx).contains(nodeOffset)) {
+    auto listsUpdatesForNodeOffset = getListsUpdatesForNodeOffsetIfExists(listFileID, nodeOffset);
+    if (listsUpdatesForNodeOffset == nullptr) {
         return false;
     }
-    return !listUpdatesPerChunk.at(chunkIdx).at(nodeOffset).deletedRelIDs.empty();
+    return !listsUpdatesForNodeOffset->deletedRelIDs.empty();
+}
+
+void ListsUpdatesStore::updateRelIfNecessary(const shared_ptr<ValueVector>& srcNodeIDVector,
+    const shared_ptr<ValueVector>& dstNodeIDVector, const ListsUpdateInfo& listsUpdateInfo) {
+    auto srcNodeID = srcNodeIDVector->getValue<nodeID_t>(
+        srcNodeIDVector->state->selVector->selectedPositions[0]);
+    auto dstNodeID = dstNodeIDVector->getValue<nodeID_t>(
+        dstNodeIDVector->state->selVector->selectedPositions[0]);
+    bool insertUpdatedRel = true;
+    for (auto direction : REL_DIRECTIONS) {
+        auto boundNodeID = direction == FWD ? srcNodeID : dstNodeID;
+        if (listsUpdatesPerTablePerDirection[direction].contains(boundNodeID.tableID)) {
+            // The rel that we are going to update either stored in persistentStore or updateStore.
+            if (listsUpdateInfo.isStoredInPersistentStore()) {
+                // If the rel is stored in persistentStore, we should store the update in the
+                // updatedRels, and store the <list_offset_t, ft_tuple_idx> index pair inside the
+                // updatedPersistentListOffsets of the boundNode.
+                if (insertUpdatedRel) {
+                    listsUpdates.at(listsUpdateInfo.propertyID)
+                        ->append(vector<shared_ptr<ValueVector>>{listsUpdateInfo.propertyVector});
+                    insertUpdatedRel = false;
+                }
+                getOrCreateListsUpdatesForNodeOffset(direction, boundNodeID)
+                    ->updatedPersistentListOffsets.at(listsUpdateInfo.propertyID)
+                    .updateEntry(direction == FWD ? listsUpdateInfo.fwdListOffset :
+                                                    listsUpdateInfo.bwdListOffset,
+                        listsUpdates.at(listsUpdateInfo.propertyID)->getNumTuples() - 1);
```

Comment:
Write in comments /* ftTupleIdx */ after `listsUpdateInfo.propertyID)->getNumTuples() - 1`, i.e., `listsUpdateInfo.propertyID)->getNumTuples() - 1 /* ftTupleIdx */`.

Reply:
done